src/components/Image/index.tsx

```diff
@@ -37,29 +37,59 @@ function Image({source: propsSource, isAuthTokenRequired = false, session, onLoa
         },
         [onLoad, updateAspectRatio],
     );
+
+    /**
+     * trying to figure out if the current session is expired or fresh from a necessary reauthentication
+     */
+    const previousSessionAge = useRef<number | undefined>();
```

Comment:
We have a hook called `usePrevious`, let use it.

Reply:
ok i'll check that